compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -14417,7 +14413,8 @@ private STNode parseErrorArgListMatchPattern(ParserRuleContext context) {
                 STNode variableName = createCaptureOrWildcardBP(parseVariableName());
                 return STNodeFactory.createTypedBindingPatternNode(varType, variableName);
             case CLOSE_PAREN_TOKEN:
-                return null;
+                return SyntaxErrors.createMissingTokenWithDiagnostics(SyntaxKind.IDENTIFIER_TOKEN,
+                        DiagnosticErrorCode.ERROR_MISSING_MATCH_PATTERN);
```

Comment:
Shall we use a separate diagnostic error code here since any match pattern is not a valid grammar here? 
May be `ERROR_MISSING_ERROR_MSG_MATCH_PATTERN` or `ERROR_MISSING_SIMLPE_MATCH_PATTERN`

Reply:
My bad, look same method is called in couple of places. Looks OK!